### PR TITLE
#715 remove outbound shipment save button 3

### PR DESCRIPTION
**Pull request by joshxg**
- This is essentially updating all of the actual invoice fields to utilise hooks rather than the reducer/draft - but not the lines, yet.
- Couple of things that I'd prefer to fix a little later down the road: 
1. Deleting I've commented out for now. Will create a custom hook for this. It's essentially not implemented for all the pages, so I might as well leave it at this point and implement this for all pages at the same time.
2. The mock server/worker status changes aren't working well - need to check up on it, but I suspect the dates aren't being set correctly. However, changing status on the demo server is working. Since we need to address the invoice statuses (with adhoc/stocktransfers etc) anyway, I thought to bundle this problem with that instead 😛

### CODE REVIEW

File: packages/invoices/src/OutboundShipment/api/api.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
+import {
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/api/api.ts:1]: hmm. What did I break 🤔

**Comment by joshxg** [comment]
- Hopefully by now this is looking all pretty familiar!
- Next step is to change the `GeneralTab` component to be as the other packages.
- Then, moving onto the `ItemModal` and updating that to hopefully 🤞 have less props and be a little simpler

File: packages/invoices/src/OutboundShipment/DetailView/SidePanel.tsx [hunk 2]
```diff
@@ -101,45 +107,28 @@ const RelatedDocumentsSection: FC<SidePanelProps> = () => {
...
+        <DetailPanelAction
+          icon={<CopyIcon />}
+          title={t('link.copy-to-clipboard')}
+          onClick={copyToClipboard}
+        />
       }
     >
-      <AdditionalInfoSection draft={draft} />
-      <RelatedDocumentsSection draft={draft} />
+      <AdditionalInfoSection />
```
Comment (mark-prins) [inline packages/invoices/src/OutboundShipment/DetailView/SidePanel.tsx:130]: a minor thought.. the `dev.log-draft` entries could be removed from the translation files now

**Review by mark-prins — APPROVED** [review]
👍  that all looks good - thanks. yeah, starting to look familiar.. and much cleaner overall

File: packages/invoices/src/OutboundShipment/DetailView/SidePanel.tsx (hunk 2 above)
Comment (joshxg) [reply packages/invoices/src/OutboundShipment/DetailView/SidePanel.tsx:130]: Ah yes! 

I will remember to remove this - for now I'd like to resolve the conflicts and merge as at a guess I'll just cause more conflicts by touching the translation files!

**Pull request merged**